tests/accuracy/test_lm_eval_correctness.py

```diff
@@ -23,104 +23,91 @@
 lm_eval: "lm_eval_t" = pytest.importorskip("lm_eval",
                                            reason="lm_eval required")
 
-
-class Metric(TypedDict):
-    name: str
-    value: float
-
-
-class Task(TypedDict):
-    name: str
-    metrics: List[Metric]
-
-
-# to support python3.8 typing prior to adding `Required`/`NotRequired`, this
-# class stores the optional keys and the `EvalTaskDefinition` subclass inherits
-# those alongside the required keys it defines.
-class EvalTaskDefinitionOpts(TypedDict, total=False):
-    enable_tensor_parallel: bool
-    extra_args: Dict[str, Any]
-    rtol: float
+RTOL = 0.02
+TEST_DATA_FILE = os.environ.get("LM_EVAL_TEST_DATA_FILE", None)
 
 
-class EvalTaskDefinition(EvalTaskDefinitionOpts):
-    model_name: str
-    tasks: List[Task]
+def wait_for_server(timeout=900) -> bool:
 
+    def try_connection() -> bool:
+        try:
+            r = requests.get("http://localhost:8000/health")
+            return r.status_code == 200
+        except Exception as _:
+            return False
 
-TEST_DATA_FILE = os.environ.get("LM_EVAL_TEST_DATA_FILE", None)
-if TEST_DATA_FILE is None:
-    raise ValueError("LM_EVAL_TEST_DATA_FILE env variable is not set.")
-TEST_DATA_FILE = Path(TEST_DATA_FILE)
-
-TEST_DATA: List[EvalTaskDefinition] = [
-    pytest.param(eval_def, id=eval_def["model_name"])
-    for eval_def in yaml.safe_load(TEST_DATA_FILE.read_text(encoding="utf-8"))
-]
-DEFAULT_RTOL = 0.05
-
-
-@pytest.mark.parametrize("eval_data", TEST_DATA)
-def test_lm_eval_correctness(
-    eval_data: EvalTaskDefinition,
-    logger: logging.Logger,
-    monkeypatch: pytest.MonkeyPatch,
-):
-    monkeypatch.setenv("TOKENIZERS_PARALLELISM", "false")
-    monkeypatch.setenv("OPENAI_API_KEY", "dummy")
-
-    model_name = eval_data["model_name"]
-    logger.info("building server startup args")
-    vllm_args = {
-        "--model": model_name,
-        "--disable-log-requests": None,
-        "--max-model-len": 4096,
-    }
+    timeout_part = 15  # retry every 15 seconds
+    time_waited = 0
+    while time_waited <= timeout:
+        time.sleep(timeout_part)
+        if try_connection():
+            return True
+        time_waited = time_waited + timeout_part
 
-    if eval_data.get("enable_tensor_parallel") is True:
-        tp = torch.cuda.device_count()
-        logger.info("Enabling tensor parallelism with %d devices", tp)
-        vllm_args["--tensor-parallel-size"] = tp
+    return False
 
-    if extra_args := eval_data.get("extra_args"):
-        vllm_args.update(extra_args)
 
+def launch_lm_eval(eval_config):
+    os.environ["OPENAI_API_KEY"] = "dummy"
     openai_args = ",".join([
-        f"model={model_name}",
+        f"model={eval_config['model_name']}",
         "tokenizer_backend=huggingface",
         "base_url=http://localhost:8000/v1",
     ])
 
-    logger.info("launching server")
-    with ServerContext(vllm_args, logger=logger) as _:
```

Comment:
Because the `ContextManager` was using Ray to launch the tests, which was causing me a lot of issues + has been the source of several issues in upstream syncs.

For instance, prior to this, if we launched with `tp>1`, the `ServerContext` did not clean up the memory on the second gpu :)

The setup here is much cleaner and actually properly cleans up the GPU memory